caffeine/src/main/scala/scalacache/caffeine/CaffeineCache.scala

```diff
@@ -0,0 +1,89 @@
+package scalacache.caffeine
+
+import scalacache.{ LoggingSupport, Cache, Entry }
+import com.github.benmanes.caffeine.cache.{ Cache => CCache, Caffeine }
+
+import scala.concurrent.duration.Duration
+import org.joda.time.DateTime
+import com.typesafe.scalalogging.StrictLogging
+import scala.concurrent.Future
+
+/*
+ * Thin wrapper around Caffeine.
+ * Since Caffeine is in-memory and non-blocking,
+ * all operations are performed synchronously, i.e. ExecutionContext is not needed.
+ *
+ * Note: Would be nice to use Any here, but that doesn't conform to CCache's type bounds,
+ * because Any does not extend java.lang.Object.
+ */
+class CaffeineCache(underlying: CCache[String, Object])
+    extends Cache
+    with LoggingSupport
+    with StrictLogging {
+
+  /**
+   * Get the value corresponding to the given key from the cache
+   * @param key cache key
+   * @tparam V the type of the corresponding value
+   * @return the value, if there is one
+   */
+  override def get[V](key: String) = {
+    val entry = Option(underlying.getIfPresent(key).asInstanceOf[Entry[V]])
+    /*
+     Note: we could delete the entry from the cache if it has expired,
+     but that would lead to nasty race conditions in case of concurrent access.
+     We might end up deleting an entry that another thread has just inserted.
+     */
+    val result = entry.flatMap { e =>
```

Comment:
If expired, you could use the `Map.computeIfPresent()` to eagerly evict atomically. The published `RemovalCause` would be `explicit` though, unless you had some adapting scheme.